openmetadata-service/src/main/java/org/openmetadata/service/resources/elasticSearch/ElasticSearchResource.java

```diff
@@ -0,0 +1,482 @@
+              }
+            });
+        return Response.status(Response.Status.OK).entity("Reindexing Started for entity.").build();
+      }
+    }
+  }
+
+  private synchronized Response startFullReindexing(String startedBy, String entityFQN, UriInfo uriInfo)
+      throws IOException {
+    String fullReindexing = dao.entityExtensionTimeSeriesDao().getLatestExtension(entityFQN, ELASTIC_SEARCH_EXTENSION);
+    EventPublisherJob fullReindexjob = JsonUtils.readValue(fullReindexing, EventPublisherJob.class);
+    if (fullReindexjob != null && fullReindexjob.getStatus() == EventPublisherJob.Status.RUNNING) {
+      return Response.status(Response.Status.FORBIDDEN)
+          .entity("Full Reindexing is Running. Cannot issue new request.")
+          .build();
+    } else {
+      // create a new Job
+      Long startTime = Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant()).getTime();
+      EventPublisherJob newJob =
+          new EventPublisherJob()
+              .withName(String.format("ElasticSearchReindexJob:%s", entityFQN))
+              .withPublisherType(EventPublisherJob.PublisherType.ELASTIC_SEARCH)
+              .withRunMode(EventPublisherJob.RunMode.BATCH)
+              .withStatus(EventPublisherJob.Status.RUNNING)
+              .withTimestamp(startTime)
+              .withStats(new Stats().withFailed(0).withSuccess(0).withTotal(0))
+              .withStartedBy(startedBy)
+              .withStartTime(startTime);
+
+      dao.entityExtensionTimeSeriesDao()
+          .insert(entityFQN, ELASTIC_SEARCH_EXTENSION, "eventPublisherJob", JsonUtils.pojoToJson(newJob));
+
+      // Update Listener
+      elasticSearchBulkProcessorListener.setRequestIssuer(startedBy);
+      elasticSearchBulkProcessorListener.setEntityFQN(entityFQN);
+      elasticSearchBulkProcessorListener.setStartTime(startTime);
+      elasticSearchBulkProcessorListener.resetCounters();
+
+      // Start Full Reindexing
+      threadScheduler.submit(
+          () -> {
+            try {
+              updateEntity(uriInfo, TABLE);
+              updateEntity(uriInfo, TOPIC);
+              updateEntity(uriInfo, DASHBOARD);
+              updateEntity(uriInfo, PIPELINE);
+              updateEntity(uriInfo, USER);
+              updateEntity(uriInfo, TEAM);
+              updateEntity(uriInfo, GLOSSARY_TERM);
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+          });
+      return Response.status(Response.Status.OK).entity("Full Reindexing Started").build();
+    }
+  }
+
+  private synchronized void updateEntity(UriInfo uriInfo, String entityType) throws IOException {
+    elasticSearchBulkProcessorListener.allowTotalRequestUpdate();
+
+    ElasticSearchIndexDefinition.ElasticSearchIndexType indexType =
+        elasticSearchIndexDefinition.getIndexMappingByEntityType(entityType);
+    // Delete index
+    elasticSearchIndexDefinition.deleteIndex(indexType);
```

Comment:
we shouldn't be deleting any of these indexes. However take a param from user to re-create indexes then only drop and create